src/components/AssetManagementPanel/AssetManagementApplication/index.tsx

```diff
@@ -27,10 +28,34 @@ export const AssetManagementApplication = ({
   );
   const { send: sendSurrender, state: surrenderingState } = useContractFunctionHook(titleEscrow, "transferTo");
 
+  const { send: changeHolder, state: holderState } = useContractFunctionHook(titleEscrow, "changeHolder");
+
   const onSurrender = () => {
     sendSurrender(tokenRegistryAddress);
   };
 
+  const onTransfer = (nextHolder: string) => {
+    changeHolder(nextHolder);
+  };
+
+  const onSetFormAction = (AssetManagementActions: AssetManagementActions) => {
+    setAssetManagementAction(AssetManagementActions);
+    setHolderErrorState("");
+  };
+
+  useEffect(() => {
+    setHolderErrorState(holderState);
+    console.log("reseting holder error state", holderState);
+
+    if (holderState === "PENDING_CONFIRMATION") {
+      onSetFormAction(AssetManagementActions.None);
+    }
+
+    if (holderState === "CONFIRMED") {
```

Comment:
i see that you are trying to refresh the holder when the transfer is complete... maybe in this case, can abstract the whole bunch of hooks into a context so there is only one place to read from

Reply:
as discussed, will hold out on this fn and will wait for all 3 to be done before adding the refresh.